### PR TITLE
Improve Typescript type definitions

### PR DESCRIPTION
**Pull request by ygerg**
Improves Typescript type definitions, especially regarding possible styles. Hopefully got them all there now.

I also omitted the props used by the library from possible modal props, do you agree with this approach?

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/**/*.d.ts
```
Comment (ygerg) [inline .eslintignore]: Added this because of an eslint conflic, which mistakenly treats .d.ts as js and gets confused

File: index.d.ts [hunk 2]
```diff
@@ -7,30 +8,63 @@ declare module 'react-native-picker-select' {
...
+        | 'testID'
+        | 'visible'
+        | 'transparent'
+        | 'animationType'
+        | 'supportedOrientations'
+        | 'onDismiss'
+        | 'onOrientationChange'
+    >;
+
+    export interface PickerSelectProps {
```
Comment (ygerg) [inline index.d.ts]: This is a breaking change to avoid name clash with React Native `PickerProps`

Comment (ygerg) [reply index.d.ts]: Another alternative is to change the import to `PickerProps as RNPickerProps` and keep the old naming scheme

**Comment by lfkwtz** [comment]
Thanks for this - I’m not very familiar with typescript, so I’ll need another set of eyes on this to review.

**Comment by ygerg** [comment]
Something that crossed my mind today was that maybe the whole `declare module` should be removed as it's not really useful here and I don't see many libraries doing that.

Also, I think we should add `declare class Picker extends React.Component<PickerSelectProps> {}`

**Comment by ygerg** [comment]
Committed those changes; removed `declare module` altogether and added `declare class Picker`

**Pull request merged**